mobile/src/people/people-department.tsx

```diff
@@ -51,7 +47,12 @@ export class PeopleDepartmentImpl extends React.Component<PeopleDepartmentProps
     }
 
     public render() {
-        return <EmployeesList employees={this.props.employees.employeesById.toArray().filter(this.props.employeesPredicate)} onItemClicked={this.props.onItemClicked} />;
+        return <EmployeesList employees={this.props.employees.employeesById.toArray().filter(this.employeesPredicate)} onItemClicked={this.props.onItemClicked} />;
+    }
+
+    private employeesPredicate = (employee: Employee) => {
+        return employee.name.startsWith(this.props.filter) &&
+            this.props.userEmployee && employee.departmentId === this.props.userEmployee.departmentId;
```

Comment:
Ok. I think it was a bug. shouldComponentUpdate used current predicate (`this.props.predicate`) in both `props.employees` and `nextProps.employees` cases. It should've used `nextProps.predicate` instead. Here, in this code, you repeat the same bug. You probably want to pass props into this function instead (and make it static since it doesn't depend on `this` anymore)